### PR TITLE
move testGitOpsConnection mutation to go

### PR DESCRIPTION
**Pull request by salahalsaleh**

### CODE REVIEW

File: kotsadm/pkg/handlers/gitops.go [hunk 1]
```diff
@@ -105,6 +110,129 @@ func DisableAppGitOps(w http.ResponseWriter, r *http.Request) {
...
+		logger.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	go func(appID, clusterID, appSlug, appName, downstreamName string) {
+		currentVersion, err := downstream.GetCurrentVersion(appID, clusterID)
+		if err != nil {
+			logger.Error(err)
+			w.WriteHeader(http.StatusInternalServerError)
```
Comment (divolgin) [inline kotsadm/pkg/handlers/gitops.go:183]: you can't write back to the client in a go routine.   when this runs, the HTTP request has already been completed

Comment (salahalsaleh) [reply kotsadm/pkg/handlers/gitops.go:183]: oh, good catch, i had this outside of the go routine, then i wrapped it and forgot about this

File: kotsadm/pkg/handlers/gitops.go [hunk 2]
```diff
@@ -105,6 +110,129 @@ func DisableAppGitOps(w http.ResponseWriter, r *http.Request) {
...
+			defer os.RemoveAll(pendingVersionArchive)
+
+			_, err = gitops.CreateGitOpsCommit(downstreamGitOps, appSlug, appName, int(pendingVersion.ParentSequence), pendingVersionArchive, downstreamName)
+			if err != nil {
+				logger.Error(err)
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+		}
+	}(a.ID, d.ClusterID, a.Slug, a.Name, d.Name)
```
Comment (divolgin) [inline kotsadm/pkg/handlers/gitops.go:231]: it's legal to use `a` inside the go routine.  you don't need to pass these in as arguments.  your call on this one.

File: kotsadm/pkg/handlers/gitops.go [hunk 3]
```diff
@@ -105,6 +110,129 @@ func DisableAppGitOps(w http.ResponseWriter, r *http.Request) {
...
+	}
+
+	if err := gitops.SetGitOpsError(a.ID, d.ClusterID, ""); err != nil {
+		logger.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	go func(appID, clusterID, appSlug, appName, downstreamName string) {
+		currentVersion, err := downstream.GetCurrentVersion(appID, clusterID)
```
Comment (divolgin) [inline kotsadm/pkg/handlers/gitops.go:180]: This should work like checking for updates works or how setting registry credentials works.  even if we can't show a status update, a second request to this handler should fail if the first one is still running

**Pull request merged**